### PR TITLE
Removing local registry from the integration scripts

### PR DESCRIPTION
**Pull request by yue9944882**
/cc @ryanjbaxter

### CODE REVIEW

**Comment by ryanjbaxter** [comment]
Could you resolve the merge conflicts?

**Comment by yue9944882** [comment]
@ryanjbaxter i will rewrite the discovery integration tests based on this in a follow-up. this pull majorly refactors the integration test script and removes the dependency for local registry during tests..

**Comment by yue9944882** [comment]
> https://hub.docker.com/r/springcloud/spring-cloud-kubernetes-configuration-watcher/tags?page=1&ordering=last_updated

@ryanjbaxter btw i was not able to find sources for image "springcloud/spring-cloud-kubernetes-configuration-watcher", it's public on the dockerhub tho?

File: spring-cloud-kubernetes-integration-tests/run.sh [hunk 1]
```diff
@@ -56,81 +65,23 @@ install_kind_release() {
...
-	done
+    "${KIND}" create cluster --config=kind-config.yaml --loglevel=debug
 
     # set KUBECONFIG to point to the cluster
     kubectl cluster-info --context kind-kind
 
 	#setup nginx ingress
 #    kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/deploy/static/provider/kind/deploy.yaml
     kubectl apply -fhttps://raw.githubusercontent.com/kubernetes/ingress-nginx/12150e318b972a03fb49d827e6cabb8ef62247ef/deploy/static/provider/kind/deploy.yaml
+    sleep 5 # hold 5 sec so that the pods can be created
```
Comment (ryanjbaxter) [inline spring-cloud-kubernetes-integration-tests/run.sh:84]: This is what the `kubectl wait` command is for, I dont see why we need to sleep

File: spring-cloud-kubernetes-integration-tests/run.sh [hunk 2]
```diff
@@ -142,15 +93,20 @@ main() {
...
-    # TODO: invoke your tests here
-    cd spring-cloud-kubernetes-core-k8s-client-it
-    ../../mvnw clean install -P it
-    cd ../
-     cd spring-cloud-kubernetes-client-config-it
-    ../../mvnw clean install -P it
-    cd ../
-    cd spring-cloud-kubernetes-configuration-watcher-it
-    ../../mvnw clean install -P it
+	WATCHER_IMAGE="docker.io/springcloud/spring-cloud-kubernetes-configuration-watcher:${MVN_VERSION}"
```
Comment (ryanjbaxter) [inline spring-cloud-kubernetes-integration-tests/run.sh:96]: Maybe we have a way of loading additional images for certain tests?

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/test/resources/spring-cloud-kubernetes-client-config-it-deployment.yaml [hunk 3]
```diff
@@ -14,16 +14,16 @@ spec:
...
       containers:
         - name: spring-cloud-kubernetes-client-config-it
-          image: localhost:5000/spring-cloud-kubernetes-client-config-it:2.0.0-SNAPSHOT
+          image: docker.io/springcloud/spring-cloud-kubernetes-client-config-it:2.0.0-SNAPSHOT
           imagePullPolicy: IfNotPresent
           readinessProbe:
             httpGet:
               port: 8080
-              path: /actuator/health/readiness
+              path: /actuator/health
```
Comment (ryanjbaxter) [inline spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/test/resources/spring-cloud-kubernetes-client-config-it-deployment.yaml:22]: Why are you changing the readiness and liveness endpoints?

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/src/test/java/org/springframework/cloud/kubernetes/configuration/watcher/ActuatorRefreshIT.java [hunk 4]
```diff
@@ -104,22 +98,14 @@
 	@Before
 	public void setup() throws Exception {
 		this.client = Config.defaultClient();
+		this.client.setHttpClient(this.client.getHttpClient().newBuilder().readTimeout(Duration.ofSeconds(15)).build());
```
Comment (ryanjbaxter) [inline spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/src/test/java/org/springframework/cloud/kubernetes/configuration/watcher/ActuatorRefreshIT.java:101]: why this change?

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]

**Comment by ryanjbaxter** [comment]
@yue9944882 the configuration watcher is located here https://github.com/spring-cloud/spring-cloud-kubernetes/tree/master/spring-cloud-kubernetes-controllers/spring-cloud-kubernetes-configuration-watcher

File: spring-cloud-kubernetes-integration-tests/run.sh (hunk 1 above)
Comment (yue9944882) [reply spring-cloud-kubernetes-integration-tests/run.sh:84]: kubectl wait will exit immediately if there's no pod present in the cluster matching the label selector. this will be reproducible in a slower development environment. my old mac hardly bears a kind cluster sorry :(

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/src/test/java/org/springframework/cloud/kubernetes/configuration/watcher/ActuatorRefreshIT.java (hunk 4 above)
Comment (yue9944882) [reply spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/src/test/java/org/springframework/cloud/kubernetes/configuration/watcher/ActuatorRefreshIT.java:101]: similar reason to https://github.com/spring-cloud/spring-cloud-kubernetes/pull/677#discussion_r534229831. the default read timeout is 5 sec in the okhttp client, in a slower development environment the kind cluster can take longer time to response the test suite..

File: spring-cloud-kubernetes-integration-tests/run.sh (hunk 1 above)
Comment (ryanjbaxter) [reply spring-cloud-kubernetes-integration-tests/run.sh:84]: OK that makes sense thanks

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/src/test/java/org/springframework/cloud/kubernetes/configuration/watcher/ActuatorRefreshIT.java (hunk 4 above)
Comment (ryanjbaxter) [reply spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/src/test/java/org/springframework/cloud/kubernetes/configuration/watcher/ActuatorRefreshIT.java:101]: We should probably extract this code to create the client in a common place so it is not repeated in every test.

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/test/resources/spring-cloud-kubernetes-client-config-it-deployment.yaml (hunk 3 above)
Comment (yue9944882) [reply spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/test/resources/spring-cloud-kubernetes-client-config-it-deployment.yaml:22]: reverted

File: spring-cloud-kubernetes-integration-tests/run.sh (hunk 2 above)
Comment (yue9944882) [reply spring-cloud-kubernetes-integration-tests/run.sh]: declared the image names as an array variables

Comment (ryanjbaxter) [reply spring-cloud-kubernetes-integration-tests/run.sh]: sounds good

File: spring-cloud-kubernetes-integration-tests/run.sh [hunk 5]
```diff
@@ -142,15 +97,21 @@ main() {
...
-    cd spring-cloud-kubernetes-core-k8s-client-it
-    ../../mvnw clean install -P it
-    cd ../
-     cd spring-cloud-kubernetes-client-config-it
-    ../../mvnw clean install -P it
-    cd ../
-    cd spring-cloud-kubernetes-configuration-watcher-it
-    ../../mvnw clean install -P it
+	for i in "${PULLING_IMAGES[@]}"; do
+		docker pull $i
```
Comment (ryanjbaxter) [inline spring-cloud-kubernetes-integration-tests/run.sh:101]: A little logging here about which images we are pulling and loading would be useful I think

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/test/resources/spring-cloud-kubernetes-client-config-it-deployment.yaml (hunk 3 above)
Comment (ryanjbaxter) [reply spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/test/resources/spring-cloud-kubernetes-client-config-it-deployment.yaml:22]: Doesnt look like it was reverted

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]
Couple of minor things

File: spring-cloud-kubernetes-integration-tests/run.sh (hunk 5 above)
Comment (yue9944882) [reply spring-cloud-kubernetes-integration-tests/run.sh:102]: done

File: spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/test/resources/spring-cloud-kubernetes-client-config-it-deployment.yaml (hunk 3 above)
Comment (yue9944882) [reply spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-client-config-it/src/test/resources/spring-cloud-kubernetes-client-config-it-deployment.yaml:22]: ah.. oversight.. plz take another look
